modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -90,20 +89,22 @@ const reduceInlineTextElements = (editor: Editor, merge: boolean | undefined): v
 
     Tools.each(dom.select('*[data-mce-fragment]'), (node) => {
       const isInline = Type.isNonNullable(textInlineElements[node.nodeName.toLowerCase()]);
-      if (isInline && StyleUtils.hasInheritableStyles(dom, node)) {
-        for (let parentNode = node.parentElement; Type.isNonNullable(parentNode) && parentNode !== root; parentNode = parentNode.parentElement) {
-          // Check if the parent has a style conflict that would prevent the child node from being safely removed,
-          // even if a exact node match could be found further up the tree
-          const styleConflict = StyleUtils.hasStyleConflict(dom, node, parentNode);
-          if (styleConflict) {
-            break;
+      if (isInline) {
+        // Remove nodes recursively if identical to any parent node
+        const stripNodes = (currentNode: HTMLElement) => {
+          // If node has only one child, that child should be removed first
+          // Only begin removing nodes when currentNode has 0 or >1 children
+          if (currentNode.childNodes.length === 1) {
+            stripNodes(currentNode.childNodes.item(0) as HTMLElement);
           }
-
-          if (elementUtils.compare(parentNode, node)) {
-            dom.remove(node, true);
-            break;
+          // Check recursively if the current node has the same attributes and styles as any parent
+          const conflictWithParent = (parentNode: HTMLElement | null): boolean => Type.isNonNullable(parentNode) && parentNode !== root
+            && (elementUtils.compare(currentNode, parentNode) || conflictWithParent(parentNode.parentElement));
```

Comment:
Could we add more tests for this not just styles since compare will not merge any similar element that is a textInlineElement so strong, b, mark etc also any attribute so spans with data attributes only or classes etc. I think that is a good thing to reduce those but we should have tests for that since if we ever refactor this then things might break. It would also be good to just have a test that verifies that `a` elements are not touched even if they have styles etc. Right now they wouldn't be since for one it's not a textInlineElement and the compare function would see that the element name is not the same as the parent and links can't be nested. But since this is such a central feature of the editor I would want that to be covered as well.